lib-gphotos/client.go

```diff
@@ -102,29 +104,49 @@ func (client *Client) UploadFile(filePath string, pAlbumID ...string) (*photosli
 		return nil, stacktrace.Propagate(err, "failed getting uploadToken for %s", filename)
 	}
 
-	batchResponse, err := client.MediaItems.BatchCreate(&photoslibrary.BatchCreateMediaItemsRequest{
-		AlbumId: albumID,
-		NewMediaItems: []*photoslibrary.NewMediaItem{
-			&photoslibrary.NewMediaItem{
-				Description:     filename,
-				SimpleMediaItem: &photoslibrary.SimpleMediaItem{UploadToken: uploadToken},
+	retry := true
+	retryCount := 0
+	for retry != false {
+		retry = false
+		batchResponse, err := client.MediaItems.BatchCreate(&photoslibrary.BatchCreateMediaItemsRequest{
+			AlbumId: albumID,
+			NewMediaItems: []*photoslibrary.NewMediaItem{
+				&photoslibrary.NewMediaItem{
+					Description:     filename,
+					SimpleMediaItem: &photoslibrary.SimpleMediaItem{UploadToken: uploadToken},
+				},
 			},
-		},
-	}).Do()
-	if err != nil {
-		return nil, stacktrace.Propagate(err, "failed adding media %s", filename)
-	}
+		}).Do()
+		if err != nil {
+			// handle rate limit error by sleeping and retrying
+			if strings.HasPrefix(err.Error(), "googleapi: Error 429") {
```

Comment:
Instead of parsing error it's better to use HTTP Server response:
```
if batchResponse.ServerResponse.HTTPStatusCode == 429 {
```

Reply:
thanks